browser/resources/settings/brave_sync_page/brave_sync_subpage.ts

```diff
@@ -187,87 +201,54 @@ class SettingBraveSyncSubpage extends SettingBraveSyncSubpageBase {
           StatusAction.RETRIEVE_TRUSTED_VAULT_KEYS))
   }
 
-  /**
-  * @return {boolean}
-  * @private
-  */
-  computeSyncDisabledByAdmin_() {
-    return this.syncStatus != undefined && !!this.syncStatus.managed
+  private computeSyncDisabledByAdmin_(): boolean {
+    return this.syncStatus !== undefined && !!this.syncStatus.managed
   }
 
-  /**
-  * @return {boolean}
-  * @private
-  */
-  computeLockedSafeStorage_() {
-    return this.syncStatus != undefined && !!this.syncStatus.hasSyncWordsDecryptionError
-      && !this.syncStatus.isOsEncryptionAvailable
+  private computeLockedSafeStorage_(): boolean {
+    return this.syncStatus !== undefined &&
+           !!this.syncStatus.hasSyncWordsDecryptionError &&
+           !this.syncStatus.isOsEncryptionAvailable
   }
 
-  /**
-  * @return {boolean}
-  * @private
-  */
-  computeUnlockedSafeStorageCannotDecryptSeed_() {
-    return this.syncStatus != undefined &&
+  private computeUnlockedSafeStorageCannotDecryptSeed_(): boolean {
+    return this.syncStatus !== undefined &&
       !!this.syncStatus.hasSyncWordsDecryptionError &&
       !!this.syncStatus.isOsEncryptionAvailable
   }
 
-  /** @protected */
-  currentRouteChanged() {
-    const router = Router.getInstance();
-    if (router.getCurrentRoute() == router.getRoutes().BRAVE_SYNC_SETUP) {
-      this.onNavigateToPage_()
-      return
-    }
-
-    if (router.getRoutes().BRAVE_SYNC_SETUP.contains(router.getCurrentRoute())) {
-      return
-    }
-
-    this.onNavigateAwayFromPage_()
-  }
-
-  /**
-  * @param {!PageStatus} expectedPageStatus
-  * @return {boolean}
-  * @private
-  */
-  isStatus_(expectedPageStatus) {
-    return expectedPageStatus == this.pageStatus_;
+  private isStatus_(expectedPageStatus: PageStatus): boolean {
+    return expectedPageStatus === this.pageStatus_
   }
 
-  /** @private */
-  onNavigateToPage_() {
+  private onNavigateToPage_() {
     const router = Router.getInstance()
-    assert(router.getCurrentRoute() == router.getRoutes().BRAVE_SYNC_SETUP)
+    assert(router.getCurrentRoute() === router.getRoutes().BRAVE_SYNC_SETUP)
     if (this.beforeunloadCallback_) {
       return
     }
 
     // Triggers push of prefs to our handler
     this.browserProxy_.didNavigateToSyncPage()
 
-    this.beforeunloadCallback_ = event => {
+    this.beforeunloadCallback_ = (event: Event) => {
```

Comment:
type should be implied by the definition on the class
```suggestion
    this.beforeunloadCallback_ = event => {
```

Reply:
Nice, fixed here https://github.com/brave/brave-core/pull/27107/commits/b95098f0760a9a3dc6732b661bf8a5184b2c03f9